var/spack/repos/builtin/packages/py-pkgconfig/package.py

```diff
@@ -20,7 +20,7 @@ class PyPkgconfig(PythonPackage):
     depends_on("python@3.3:", when="@1.5.5:", type=("build", "run"))
     depends_on("python@2.7,3.3:", when="@1.5:", type=("build", "run"))
     depends_on("python@2.6:2.7,3.3:", when="@1.4:", type=("build", "run"))
-    depends_on("py-poetry-core@1:", when="@1.5.5:", type="build")
-    depends_on("py-poetry-core@0.12:", when="@1.5:", type="build")
+    depends_on("py-poetry@1:", when="@1.5.2:", type="build")
+    depends_on("py-poetry@0.12:", when="@1.5:", type="build")
```

Comment:
This is incorrect

Reply:
I kept getting "No module named poetry" error while installing this, but I'll test it again without this

Reply:
@adamjstewart I'm having the same issue when re-building. I think it's a similar problem to https://github.com/spack/spack/issues/29308 Here is the build log:

```
==> py-pkgconfig: Executing phase: 'install'
==> [2022-10-01-10:45:23.400953] '/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/python-3.9.13-5zn3hui4ysxknqhfvmbsupmf5ysefebn/bin/python3.9' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6' '.'i.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-v265qa90
Created temporary directory: /tmp/pip-build-tracker-rrrnkivo
Initialized build tracking at /tmp/pip-build-tracker-rrrnkivo
Created build tracker: /tmp/pip-build-tracker-rrrnkivo
Entered build tracker: /tmp/pip-build-tracker-rrrnkivo
Created temporary directory: /tmp/pip-install-jbh6ib2n
Processing /home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src
  Added file:///home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src to build tracker '/tmp/pip-build-tracker-rrrnkivo'
  Created temporary directory: /tmp/pip-modern-metadata-ge9ca6vp
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml): finished with status 'done'
ERROR: Exception:
Traceback (most recent call last):
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper
    status = run_func(*args)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 341, in run
    requirement_set = resolver.resolve(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 75, in resolve
    collected = self.factory.collect_root_requirements(root_reqs)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 506, in collect_root_requirements
    req = self._make_requirement_from_install_req(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 468, in _make_requirement_from_install_req
    cand = self._make_candidate_from_link(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 215, in _make_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 291, in __init__
    super().__init__(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 161, in __init__
    self.dist = self._prepare()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 230, in _prepare
    dist = self._prepare_distribution()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 302, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 428, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 497, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/prepare.py", line 58, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/distributions/sdist.py", line 61, in prepare_distribution_metadata
    self.req.prepare_metadata()
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/req/req_install.py", line 526, in prepare_metadata
    self.metadata_directory = generate_metadata(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/operations/build/metadata.py", line 35, in generate_metadata
    distinfo_dir = backend.prepare_metadata_for_build_wheel(metadata_dir)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_internal/utils/misc.py", line 706, in prepare_metadata_for_build_wheel
    return super().prepare_metadata_for_build_wheel(
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 188, in prepare_metadata_for_build_wheel
    return self._call_hook('prepare_metadata_for_build_wheel', {
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/wrappers.py", line 332, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-pip-22.1.2-frngqanail6q44luijakcnmb3iox3fws/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 89, in _build_backend
    obj = import_module(mod_path)
  File "/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/python-3.9.13-5zn3hui4ysxknqhfvmbsupmf5ysefebn/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'poetry'

Removed file:///home/lukedt/Documents/GitHub/spack/var/spack/stage/spack-stage-py-pkgconfig-1.5.5-yzezoqcuwxhqkkc4z6o7purymq3ufhe6/spack-src from build tracker '/tmp/pip-build-tracker-rrrnkivo'
Removed build tracker: '/tmp/pip-build-tracker-rrrnkivo'
```

Reply:
Can you share `spack-built-env.txt` and also the output of `python3 -m sysconfig`?

Reply:
Are you using Spack-installed Python or the system Python as an external package?

Reply:
This system has python installed separately, yes.

[spack-build-env.txt](https://github.com/spack/spack/files/9693183/spack-build-env.txt)
[python3_sysconfig.txt](https://github.com/spack/spack/files/9693186/python3_sysconfig.txt)

Reply:
Okay, so you're using Spack-installed Python. So it's not an issue with sysconfig/distutils hacking.

Another possibility is that this is related to #32375. Can you run:
```console
$ tree /home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-poetry-core-1.2.0-xr4smvmirljhej6hnwtnsb5bmlrnyg6b/lib/python3.9/site-packages
```
(or `find` if `tree` is not installed)

I'm guessing that the entire `poetry` directory and all of its contents will be missing. Do you have a global `.gitignore` somewhere? Can you upload that?

Reply:
```
/home/lukedt/Documents/GitHub/spack/opt/spack/linux-fedora36-zen/gcc-8.5.0/py-poetry-core-1.2.0-xr4smvmirljhej6hnwtnsb5bmlrnyg6b/lib/python3.9/site-packages
└── poetry_core-1.2.0.dist-info
    ├── direct_url.json
    ├── INSTALLER
    ├── LICENSE
    ├── METADATA
    ├── RECORD
    ├── REQUESTED
    └── WHEEL

1 directory, 7 files

```
I don't have a global `.gitignore`, just the one in `~/Documents/GitHub/spack`

Reply:
Must be pulling it in from somewhere else then. I'll see if I can hack together a solution since you're the 3rd person who's reported this issue.

Reply:
thanks for the assistance!

Reply:
so oddly enough, my latest commit ([e64fd01](https://github.com/spack/spack/pull/32936/commits/e64fd01bbf0c76b5aa145751ec999e167eafefa2)) builds OK on my machine without this error...